src/Microsoft.AspNet.Mvc.TagHelpers/Internal/AttributeMatcher.cs

```diff
@@ -88,7 +88,7 @@ public static class AttributeMatcher
             {
                 if (!context.AllAttributes.ContainsKey(attribute) ||
                     context.AllAttributes[attribute] == null ||
-                    string.IsNullOrWhiteSpace(context.AllAttributes[attribute] as string))
+                    string.IsNullOrWhiteSpace(context.AllAttributes[attribute].ToString()))
```

Comment:
Now that we have a bool html attribute, `as string` would return `null` whereas `ToString()` will return `true` or `false`.